node/src/store.rs

```diff
@@ -121,12 +126,25 @@ pub trait Store: Send + Sync + Debug {
     /// Returns true if height exists in the store.
     async fn has_at(&self, height: u64) -> bool;
 
-    /// Append single header maintaining continuity from the genesis to the head.
-    ///
-    /// # Note
-    ///
-    /// This method does not validate or verify that `header` is indeed correct.
-    async fn append_single_unchecked(&self, header: ExtendedHeader) -> Result<()>;
+    /// Insert single header into the store
+    async fn append_single_unchecked(&self, header: ExtendedHeader) -> Result<()> {
+        self.insert([header].into()).await
+    }
+
+    /// Insert single header into the store, validating it against neightbours, if present in store
+    async fn append_single(&self, header: ExtendedHeader) -> Result<()> {
+        self.insert([header].into()).await
+    }
+
+    /// Insert list of headers, validating that the inserted range is contiguous
+    async fn append_unchecked(&self, headers: Vec<ExtendedHeader>) -> Result<()> {
+        self.insert(headers.try_into()?).await
+    }
+
+    /// Insert list of headers validating them against neighbouring headers if they exist in store
+    async fn append(&self, headers: Vec<ExtendedHeader>) -> Result<()> {
+        self.insert(headers.try_into()?).await
+    }
```

Comment:
Remove all of them and use `TryInto<VerifiedExtentedHeaders>` at `insert` parameters